dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java

```diff
@@ -197,12 +195,20 @@ private static void fullPreorder(Stack<DependencyNode> path, DependencyNode curr
       if (!"system".equals(child.getDependency().getScope())) {
         try {
           child = resolveCompileTimeDependencies(child.getArtifact());
+          // somehow we've got an infinite recursion here
```

Comment:
Should we throw?

Reply:
I thought about this, but my gut is no. This actually happens in our graph when we include contrib and/or logback which is why I dropped them out. But I don't want to fail completely and not process the remaining items in this case. Hence the short circuit here.